token/services/ttx/endorse.go

```diff
@@ -745,6 +734,35 @@ func (s *endorseView) requestsToBeSigned() ([]*token.Transfer, error) {
 				res = append(res, transfer)
 			}
 		}
+		for _, sender := range transfer.ExtraSigners {
+			if _, err := s.tx.TokenService().SigService().GetSigner(sender); err == nil {
+				res = append(res, transfer)
+			}
+		}
 	}
 	return res, nil
 }
+
+func (s *endorseView) receiveTransaction(context view.Context) error {
+	if logger.IsEnabledFor(zapcore.DebugLevel) {
+		logger.Debugf("Receive transaction with envelope...")
+	}
+	// TODO: this might also happen multiple times because of the pseudonym. Avoid this by identity resolution at the sender
+	tx, err := ReceiveTransaction(context)
+	if err != nil {
+		return errors.Wrapf(err, "failed receiving transaction")
+	}
+	agent := metrics.Get(context)
+	agent.EmitKey(0, "ttx", "received", "env", tx.ID())
+
+	// Check the envelope exists
+	if logger.IsEnabledFor(zapcore.DebugLevel) {
+		logger.Debugf("Processes Fabric Envelope with ID [%s]", tx.ID())
+	}
+	env := tx.Payload.Envelope
+	if env == nil {
+		return errors.Errorf("expected fabric envelope")
+	}
+	s.tx.Payload.Envelope = tx.Payload.Envelope
```

Comment:
I assume since each view is initialized from a view factory, there is no issue in concurrency because this `s.tx.Payload.Envelope` assignment acts only on a single dedicated view instance correct?

Reply:
right